### PR TITLE
Twilio dev

### PR DESCRIPTION
**Pull request by Flatcova**
add the Twilio for SMS,

add dotenv for creating .env values and add Const values

integrate node-cache for the validations when checking for values between iterations

please check the logic on the validation with cache 👍

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,50 +1,89 @@
...
-      );
+    const listOfAvailability = data.locations.filter(
+      (stores) => stores.city === "LAREDO" && stores.openAppointmentSlots == 0
+    );
 
     if (listOfAvailability.length) {
       notify(listOfAvailability);
     } else {
-      console.log('COVID is winning.... 🦠 🦠 🦠 🦠 🦠');
+      console.log("COVID is winning.... 🦠 🦠 🦠 🦠 🦠");
```
Comment (a-covar) [inline index.js:81]: Is always better to use single quotes we might neet to add a jslint for this

File: index.js [hunk 2]
```diff
@@ -1,50 +1,89 @@
...
+    };
+    console.log(`🧬 💉 : ${available}`);
+
+    notifier.notify(message);
+    console.log("Twilio SMS");
+    twilio(available);
+  }
 }
 
+function twilio(locations) {
```
Comment (a-covar) [inline index.js:52]: we should extract this ot its own module just to make more clear

Comment (a-covar) [reply index.js:52]: also make it an async func

File: index.js [hunk 3]
```diff
@@ -1,50 +1,89 @@
...
-  const message = {
-    title: 'COVID Vaccine available!!! ',
-    message: available,
-    icon: ' 🧬 💉', // Absolute path (doesn't work on balloons)
-    sound: true, // Only Notification Center or Windows Toasters
-  };
-  console.log(`🧬 💉 : ${available}`);
-
-  notifier.notify(message);
+  const valores = arrayOfLocations.filter((location) => {
```
Comment (a-covar) [inline index.js]:
```suggestion
  const available = arrayOfLocations.filter((location) => {
```

File: index.js [hunk 4]
```diff
@@ -1,50 +1,89 @@
...
+    const value = myCache.get(location.storeNumber);
+    if (value !== undefined) {
```
Comment (a-covar) [inline index.js]:
```suggestion
    if (value !== undefined && value.spots !== location.openAppointmentSlots) {
```

Simplify the if statement

File: index.js (hunk 3 above)
Comment (a-covar) [inline index.js]: we should also do a seperate function just for this and name it `isInCache`. ... Also when do you clear the cache if the store is no longer available ?

**Review by a-covar — CHANGES_REQUESTED** [review]
Looks ok ... I just have questions regarding logic when a store is not available 

For example 

store A - > 1 ----- cache A - 1 
store A -> 0 ------ cache A- 1

File: index.js (hunk 1 above)
Comment (Flatcova) [reply index.js:81]: I use prettier for this kind of corrections

Comment (a-covar) [reply index.js:81]: you might need to fix the config to use single quote

File: index.js (hunk 3 above)
Comment (Flatcova) [reply index.js]: yeap I forgot to change the variable name

Comment (Flatcova) [reply index.js]: need to implement method when a store runs out of spots available.

File: index.js (hunk 2 above)
Comment (Flatcova) [reply index.js:52]: on change, need to clear the cache values.

**Pull request merged**